giraffe/src/utils/legend/staticLegend.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  Formatter,
+  LegendData,
+  LineLayerSpec,
+  LineData,
+  StaticLegend,
+} from '../../types'
+
+const peek = (arr: number[]): number => {
+  if (Array.isArray(arr) && arr.length >= 1) {
+    return arr[arr.length - 1]
+  }
+  return NaN
+}
+
+export const convertLineSpec = (
+  staticLegend: StaticLegend,
+  spec: LineLayerSpec,
+  getColumnFormatter: (colKey: string) => Formatter,
+  valueColumnKey: string
+): LegendData => {
+  const {valueAxis} = staticLegend
+  const valueFormatter = getColumnFormatter(valueColumnKey)
+  const mappings = spec?.columnGroupMaps?.fill?.mappings
+  const lineData: LineData = spec?.lineData
+
+  const colors = Object.values(lineData).map(line => line.fill)
+
+  const propertyName = valueAxis === 'x' ? 'xs' : 'ys'
+
+  const values = Object.values(lineData).map(line => {
+    const latestValue = peek(line[propertyName])
+    if (latestValue === latestValue) {
+      return valueFormatter(latestValue)
+    }
+    return ''
+  })
+
+  const {columnKeys} = spec.columnGroupMaps.fill
+  if (!Array.isArray(columnKeys)) {
+    return null
+  }
+
+  const valueColumn = {
+    key: valueColumnKey,
+    name: `Latest ${valueColumnKey}`,
+    type: spec.table.getColumnType(valueColumnKey),
+    colors,
+    values,
+  }
+
+  const legendColumns = columnKeys.map(key => {
+    // TODO: should be using formatter from config
```

Comment:
aren't we already using the formatter? (in the tooltip itself).

so this todo should be defunct
(please verify, and if so remove)

Reply:
it's a valid todo which we will address in the follow-up PRs for stacked lines and band chart